src/generator.cs

```diff
@@ -3029,6 +3030,9 @@ public void GeneratedCode (StreamWriter sw, int tabs)
 		for (int i=0; i < tabs; i++)
 			sw.Write ('\t');
 		sw.WriteLine ("[CompilerGenerated]");
+		for (int i = 0; i < tabs; i++)
+			sw.Write ('\t');
+		sw.WriteLine ("[LinkerOptimize]");
```

Comment:
if we add an attribute then the existing `[CompilerGenerated]` should be removed

Reply:
It's half-true, noisy and without a use case (with a new attribute).

If we want to keep tagging generated code then maybe its the new attribute that's needs to to it, e.g. `[BindingGenerated (Optimizable = true, ...)]`. It would be more accurate, than  `[CompilerGenerated]`. Also it's not really the linker that optimize (even if it's the same pipeline) wrt `[LinkerOptimize]`.

Reply:
`[BindingGenerated (Optimizable)]` looks strange when applied to manually written methods.

What about just `[Optimize]` or `[Optimizable]`)?

Reply:
yes, naming is hard :)

What I'd like is to avoid an (or limit the) explosion of semi-related attributes, even more if its being consumed by a single tool (mtouch/mmp). Maybe something similar to `[MethodImpl]` ?

The generator _could_ (if needed?) add the fact it's generated:

```
 [BindingImpl (BindingImplOptions.GeneratedCode)]
```

and we could reuse this for future improvements... (yes I do have some crazy ideas in need of attributes).

Reply:
That might work.

So we'd add this to generated code:

```
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
```

and just this to manually written code:

```
[BindingImpl (BindingImplOptions.Optimizable)]
```